common-model/src/main/java/com/sequenceiq/common/api/backup/base/BackupBase.java

```diff
@@ -0,0 +1,88 @@
+package com.sequenceiq.common.api.backup.base;
+
+import java.io.Serializable;
+
+import javax.validation.constraints.NotNull;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.sequenceiq.common.api.cloudstorage.old.GcsCloudStorageV1Parameters;
+import com.sequenceiq.common.api.cloudstorage.old.S3CloudStorageV1Parameters;
+import com.sequenceiq.common.api.cloudstorage.old.AdlsGen2CloudStorageV1Parameters;
+import com.sequenceiq.common.api.backup.doc.BackupModelDescription;
+import com.sequenceiq.common.api.backup.model.BackupCloudwatchParams;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public abstract class BackupBase implements Serializable {
+
+    @NotNull
+    @ApiModelProperty(BackupModelDescription.BACKUP_STORAGE_LOCATION)
+    private String storageLocation;
+
+    @ApiModelProperty(BackupModelDescription.BACKUP_S3_ATTRIBUTES)
+    private S3CloudStorageV1Parameters s3;
+
+    @ApiModelProperty(BackupModelDescription.BACKUP_ADLS_GEN_2_ATTRIBUTES)
+    private AdlsGen2CloudStorageV1Parameters adlsGen2;
```

Comment:
missing @Valid annotation

Reply:
I didn't add @valid for this class as AdlsGen2CloudStorageV1Parameters doesn't have validation inside.

Reply:
we can always make the code better and fix missing things ;)

Reply:
I mean AdlsGen2CloudStorageV1Parameters.java doesn't have an annotation that needs to be validated, not like the others that have annotations inside that need validation. I added @valid to the other classes you are suggesting but just not this one. Thanks!